### PR TITLE
Update with docs on macOS broker

### PR DESCRIPTION
**Pull request by localden**

### CODE REVIEW

File: msal-python-conceptual/advanced/macos-broker.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+title: Using MSAL Python with Authentication Brokers on macOS
+description: "Using authentication brokers on macOS enables you to simplify how your users authenticate with Microsoft Entra ID from your application, as well as take advantage of advanced functionality such as token binding, protecting any issued tokens from exfiltration and misuse."
+author: localden
+manager: CelesteDG
+
+ms.service: msal
+ms.subservice: msal-python
+ms.topic: conceptual
+ms.date: 02/07/2024
```
Comment (rayluo) [inline msal-python-conceptual/advanced/macos-broker.md:10]: I know where this date came from. It is a copy-and-paste error from the pre-existing `wam.md` doc. :-)

On a more serious note, we could also reuse the same `wam.md` doc by adding a new "Prerequisite of using broker on macOS" section. Besides the different prerequisites and different opt-in feature flag, the code snippets on both platforms are the same.

File: msal-python-conceptual/advanced/macos-broker.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+from msal import PublicClientApplication
+ 
+app = PublicClientApplication(
+    "CLIENT_ID",
+    authority="https://login.microsoftonline.com/common",
+    enable_broker_on_mac =True)
+```
+
+>[!IMPORTANT]
+>If you are writing a cross-platform application, you will also need to use `enable_broker_on_windows`, as outlined in the [Using MSAL Python with Web Account Manager](wam.md) article.
```
Comment (rayluo) [inline msal-python-conceptual/advanced/macos-broker.md:36]: Don't we also want to have this paragraph backported to the `wam.md`?

File: msal-python-conceptual/advanced/macos-broker.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+```text
+msauth.com.msauth.unsignedapp://auth 
+```
+
+For signed applications, the redirect URI should be:
+
+```text
+msauth.BUNDLE_ID://auth
+```
```
Comment (rayluo) [inline msal-python-conceptual/advanced/macos-broker.md:51]: @fengga mentioned it offline that a Python script might not be able to have bundle id at all. If so, that this "for signed applications" paragraph is not applicable to be put in this "MSAL Python Conceptual" document.

File: msal-python-conceptual/advanced/macos-broker.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+tag=508170375
+context=AADSTS50011 Description: (pii), Domain: MSAIMSIDOAuthErrorDomain.Error was thrown in location: Broker 
+errorCode=-51411 
+status=Response_Status.Status_Unexpected 
+```
+
+Once configured, you can call `acquire_token_interactive` to acquire a token.
+
+```python
+result = app.acquire_token_interactive(["User.ReadBasic.All"])
```
Comment (rayluo) [inline msal-python-conceptual/advanced/macos-broker.md:66]: Actually, MSAL Python still requires the same `parent_window_handle=app.CONSOLE_WINDOW_HANDLE` parameter as it does with the WAM, even though the value is merely a placeholder.

File: msal-python-conceptual/advanced/macos-broker.md (hunk 3 above)
Comment (localden) [reply msal-python-conceptual/advanced/macos-broker.md:51]: @fengga @rayluo can it _ever_ be signed? What about GUI apps?

File: msal-python-conceptual/advanced/macos-broker.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
+---
+title: Using MSAL Python with Authentication Brokers on macOS
+description: "Using authentication brokers on macOS enables you to simplify how your users authenticate with Microsoft Entra ID from your application, as well as take advantage of advanced functionality such as token binding, protecting any issued tokens from exfiltration and misuse."
```
Comment (oldalton) [inline msal-python-conceptual/advanced/macos-broker.md:3]: A few nit picks:
1. We only have a single broker on Mac, so should be using singular
2. Token binding isn't public yet on Mac and it only protects refresh tokens for now, I suggest rewording last sentence to more clearly represent the benefit

File: msal-python-conceptual/advanced/macos-broker.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+ms.author: ddelimarsky
+ms.reviewer: shermanouko, rayluo
+---
+
+# Using MSAL Python with Authentication Brokers on macOS
+
+>[!NOTE]
+>macOS authentication broker support is introduced with `msal` version 1.31.0.
+
+Using authentication brokers on macOS enables you to simplify how your users authenticate with Microsoft Entra ID from your application, as well as take advantage of advanced functionality such as **token binding**, protecting any issued tokens from exfiltration and misuse.
```
Comment (oldalton) [inline msal-python-conceptual/advanced/macos-broker.md:20]: Same as above, change brokers to broker, and change "protecting any issued tokens from exfiltration and misuse." to "protecting Entra ID refresh tokens".

File: msal-python-conceptual/advanced/macos-broker.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+```
+
+Once configured, you can call `acquire_token_interactive` to acquire a token.
+
+```python
+result = app.acquire_token_interactive(["User.ReadBasic.All"],
+                    parent_window_handle=app.CONSOLE_WINDOW_HANDLE)
+```
+
+The `parent_window_handle` parameter is required even though on macOS it is not used. For GUI applications, the login prompt location will be determined ad-hoc and currently cannot be bound to a specific window.
```
Comment (oldalton) [inline msal-python-conceptual/advanced/macos-broker.md:78]: Should we clarify that it will be used in the future for GUI apps?

File: msal-python-conceptual/advanced/macos-broker.md [hunk 8]
```diff
@@ -12,12 +12,12 @@ ms.author: ddelimarsky
...
 ---
 
-# Using MSAL Python with Authentication Brokers on macOS
+# Using MSAL Python with an Authentication Broker on macOS
 
 >[!NOTE]
 >macOS authentication broker support is introduced with `msal` version 1.31.0.
 
-Using authentication brokers on macOS enables you to simplify how your users authenticate with Microsoft Entra ID from your application, as well as take advantage of advanced functionality such as **token binding**, protecting any issued tokens from exfiltration and misuse.
+Using an authentication brokers on macOS enables you to simplify how your users authenticate with Microsoft Entra ID from your application, as well as take advantage of future functionality that protects Microsoft Entra ID tokens from exfiltration and misuse.
```
Comment (oldalton) [inline msal-python-conceptual/advanced/macos-broker.md:20]: We should either say Microsoft Entra ID refresh tokens, or not say anything at all :) Don't want someone to assume this is a promise that access tokens will also be protected
